homeassistant/components/shelly/utils.py

```diff
@@ -0,0 +1,15 @@
+"""Shelly helpers functions."""
+
+import logging
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_remove_entity_by_domain(hass, domain, unique_id):
+    """Remove entity by domain."""
+    entity_reg = await hass.helpers.entity_registry.async_get_registry()
+    for entity in entity_reg.entities.values():
+        if entity.entity_id.startswith(domain + ".") and entity.unique_id == unique_id:
```

Comment:
Entries have a domain prop:
```suggestion
        if entity.domain == domain and entity.unique_id == unique_id:
```

Reply:
Also, you should check the `entry.config_entry_id` is equal to the config entry of Shelly. You can use the `async_entries_for_config_entry` helper 

```python
from homeassistant.helpers import entity_registry

entries = entity_registry.async_entries_for_config_entry(entity_reg, config_entry_id)
```

Reply:
isn't current `unique_id` check enough ?